### PR TITLE
RGRIDT-831: Changing Data Source with Drop Down

### PR DESCRIPTION
**Pull request by glundgren93**
### What was happening
* Whenever the user switched data sources, the grid wasn’t adding new rows and data.

### What was done
* Created a new method that checks whether columns are new compared to existing ones and adds/removes them if necessary.
* Fixed eslint issues on AbstractGrid

### Checklist
* [x] tested locally
* [ ] documented the code
* [ ] clean all warnings and errors of eslint
* [ ] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/OSFramework/Grid/AbstractGrid.ts [hunk 1]
```diff
@@ -81,20 +81,84 @@ namespace OSFramework.Grid {
...
+            }
+        }
 
+        private _checkForNewColumns(): boolean {
+            const metadata = this.dataSource.getMetadata();
+            let hasColumns = false;
+            return Object.keys(metadata).some((source) => {
+                const columns = Array.from(this._columns.values()).map(
+                    (col) => col.config.binding
+                );
```
Comment (tmbp95) [inline code/src/OSFramework/Grid/AbstractGrid.ts:126]: Can we move these lines outside the "some" method?

File: code/src/OSFramework/Grid/AbstractGrid.ts [hunk 2]
```diff
@@ -81,20 +81,84 @@ namespace OSFramework.Grid {
...
+                const newColumns = Object.keys(metadata[source]);
+                hasColumns = newColumns.every((column) => {
+                    return columns.indexOf(`${source}.${column}`) !== -1;
```
Comment (tmbp95) [inline code/src/OSFramework/Grid/AbstractGrid.ts:129]: will this work when we have sources like -> "Sample_Product.Name" and "Name"?

File: code/src/OSFramework/Grid/AbstractGrid.ts (hunk 1 above)
Comment (glundgren93) [reply code/src/OSFramework/Grid/AbstractGrid.ts:126]: Yes! My bad

File: code/src/OSFramework/Grid/AbstractGrid.ts (hunk 2 above)
Comment (glundgren93) [reply code/src/OSFramework/Grid/AbstractGrid.ts:129]: No :(((

**Review by tmbp95 — APPROVED** [review]

**Review by bmartinho — APPROVED** [review]
Ready to go!

**Review by rugoncalves — APPROVED** [review]

**Pull request merged**